okhttp/src/main/kotlin/okhttp3/internal/tls/BasicCertificateChainCleaner.kt

```diff
@@ -63,7 +63,7 @@ class BasicCertificateChainCleaner(
         if (result.size > 1 || toVerify != trustedCert) {
           result.add(trustedCert)
         }
-        if (verifySignature(trustedCert, trustedCert)) {
+        if (verifySignature(trustedCert, trustedCert, result.size - 2)) {
```

Comment:
Maybe worth a comment on these values. -2 is the trusted cert and the site certificate I assume.